feat: Add send transaction logic

calintje opened the pull request:
# Background

## What does this PR do?
The current implementation of sending transactions on Solana does not check if the transaction is confirmed, but logs the transaction hash as if it is confirmed.

Since confirmation is not guaranteed, a client-side retry mechanism is included to resend the transaction every second until the lifetime of the blockhash, which is 150 blocks, expires. Assuming ~400 ms blocktimes, this amounts to 60 seconds. 

## What kind of change is this?
Features (non-breaking change which adds functionality)

## Why are we doing this? Any context or related work?
Improve transaction success on Solana. Especially 

# Documentation changes needed?

My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?
Send a transfer transaction on Devnet, or Mainnet

## Discord username
@calintje

cygaar committed:
feature: add abstract plugin

cygaar committed:
update pnpm lockfile

shakkernerd committed:
Merge pull request #1292 from ai16z/fix/pnpm-lockfile

fix: pnpm lock file

shakkernerd committed:
fix: add missing claude vertex case to handleProvider

shakkernerd committed:
Merge pull request #1293 from ai16z/patch/handle_provider

fix: add missing claude vertex case to handleProvider

twilwa committed:
output checkable variable for conditional

shakkernerd committed:
feat: types for Cache Store

shakkernerd committed:
chore: add adapter-redis package

shakkernerd committed:
feat: add cache env to env example

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
feat: add support for redis cache

shakkernerd committed:
Merge pull request #1294 from ai16z/fix/integration-tests

fix: output checkable variable for conditional

shakkernerd committed:
Merge branch 'develop' into feat/agent_redis_support

shakkernerd committed:
Merge pull request #1295 from ai16z/feat/agent_redis_support

feat: Add caching support for Redis

shakkernerd committed:
chore: bump version to 0.1.6-alpha.5

shakkernerd committed:
Merge pull request #1296 from ai16z/new_version_0.1.6-alpha5

chore: bump version to 0.1.6-alpha.5

shakkernerd committed:
Merge pull request #1290 from ai16z/develop

feat: Update main for v0.1.6-alpha.5

shakkernerd committed:
fix: remove clients from default character

shakkernerd committed:
Merge pull request #1297 from ai16z/fix/default_character

fix: remove clients from default character

shakkernerd committed:
fix: default character model to LLAMALOCAL

shakkernerd committed:
Merge pull request #1299 from ai16z/fix/default_character

fix: default character model to LLAMALOCAL

actions-user committed:
chore: update changelog

shakkernerd committed:
chore: Updated node version to 23.3.0 and pnpm version to 9.15.0

shakkernerd committed:
feat: release version 0.1.6

shakkernerd committed:
Merge pull request #1300 from ai16z/new_version_0.1.6

feat: release version 0.1.6

actions-user committed:
chore: update changelog

shakkernerd committed:
Merge pull request #1301 from ai16z/main

chore: rebase develop branch

tcm390 committed:
add compose random user function

tcm390 committed:
compose random user name before compose context

tcm390 committed:
compose random user name before compose context

tcm390 committed:
correct template

Commit:
no token needed for gaianet

update getting tokens to reflect no key needed for gaianet. Don't throw an error here.

tcm390 committed:
correct comment

mcp97 committed:
Merge pull request #1306 from suicidalgoofy/main

no token needed for gaianet

mcp97 committed:
Merge branch 'develop' into develop

mcp97 committed:
Merge pull request #1171 from nicky-ru/develop

fix: add lint script for plugin evm and fix lint errors

mcp97 committed:
Merge branch 'develop' into twitter-cookies

mcp97 committed:
Merge pull request #1288 from ChristopherTrimboli/twitter-cookies

chore: remove TWITTER_COOKIES env var

oxSaturn committed:
fix: update turbo to fix "cannot find package" error

mcp97 committed:
Merge pull request #1307 from oxSaturn/fix/cannot-find-package

fix: update turbo to fix "cannot find package" error

mcp97 committed:
Merge branch 'develop' into tcm-compose-random-user

oxSaturn committed:
fix: set default value for cache store

oxSaturn committed:
fix: support google

mcp97 committed:
Merge pull request #1308 from oxSaturn/fix/set-default-cache-store

fix: set default value for cache store

shakkernerd committed:
chore: bump agent-twitter-client version to v0.0.17

mcp97 committed:
Merge branch 'develop' into fix/support-google

mcp97 committed:
Merge pull request #1310 from oxSaturn/fix/support-google

fix: support google model.

mcp97 committed:
Merge branch 'develop' into chore/client_twitter

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #1311 from ai16z/chore/client_twitter

chore: bump agent-twitter-client version to v0.0.17

mcp97 committed:
Create codeql.yml

Enabling code scanning for security purposes

mcp97 committed:
Merge branch 'develop' into realitySpiral/codescanning

mcp97 committed:
Remove comments

mcp97 committed:
Create greetings.yml

added greeting workflow for first time contributors :)

mcp97 committed:
Merge branch 'develop' into realitySpiral/greetingworkflow

mcp97 committed:
feat: add auto PR / issue close after being stale for a certain period of time

mcp97 committed:
remove comments + update configuration

remov

mcp97 committed:
Merge branch 'develop' into realitySpiral/closeStalPRsIssues

mcp97 committed:
update numbers

oxSaturn committed:
fix: use MAX_TWEET_LENGTH from setting

mcp97 committed:
Merge pull request #1323 from oxSaturn/fix/max-tweet-length

fix: use MAX_TWEET_LENGTH from setting

ChristopherTrimboli committed:
fix twitter login notifications

mcp97 committed:
Update greetings.yml

amirkhonov committed:
fix: Add OLLAMA model to the getTokenForProvider class

ryanleecode committed:
fix: postgres adapter schema

mcp97 committed:
Merge pull request #1338 from amirkhonov/develop

fix: Add OLLAMA model to the getTokenForProvider class

mcp97 committed:
Merge branch 'develop' into fix/postgres-adapter-schema

mcp97 committed:
Merge pull request #1345 from ryanleecode/fix/postgres-adapter-schema

fix: postgres adapter schema

0x330a committed:
fix: change max cast length to the current max cast length

0x330a committed:
refactor: reference the max cast length in the content trimming process and export as a const so it's available

0x330a committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1347 from 0x330a/main

Update farcaster client max cast length

trbutler4 committed:
add documentation on pnpm node version

trbutler4 committed:
remove extra spacing

trbutler4 committed:
Merge branch 'develop' into add-note-on-pnpm-node-version

calintje committed:
Add send transaction logic

ChristopherTrimboli committed:
Merge branch 'develop' into chore/twitter-cookies

lalalune committed:
elizaOS

imwylin committed:
Update DOCUMENTATION links to point to https://elizaOS.github.io/eliza/

shakkernerd committed:
chore: ai16z -> elizaOS

madjin committed:
revert discord url

shakkernerd committed:
Merge pull request #1355 from elizaOS/elizaos_discord

chore: revert discord url

lalalune committed:
Merge pull request #1352 from elizaOS/elizaos

feat: elizaOS

lalalune committed:
Merge pull request #1356 from elizaOS/develop

chore: Merge Develop into Main

lalalune committed:
Merge branch 'main' into patch-1

lalalune committed:
Merge pull request #1353 from imwylin/patch-1

Update DOCUMENTATION links to point to https://elizaOS.github.io/eliza/

lalalune committed:
Merge branch 'develop' into add-note-on-pnpm-node-version

lalalune committed:
elizaos/core

lalalune committed:
Merge branch 'main' of http://github.com/ai16z/eliza into elizaos-core

lalalune committed:
Merge pull request #1357 from elizaOS/elizaos-core

feat: change @elizaos/eliza to @elizaos/core

lalalune committed:
Merge pull request #1359 from elizaOS/develop

chore: develop -> main, change elizaos/eliza to elizaos/core

actions-user committed:
chore: update changelog

shakkernerd committed:
chore: bump version to v.0.1.7-alpha.1

shakkernerd committed:
Merge pull request #1360 from elizaOS/new_version_0.1.7-alpha.1

chore: New version 0.1.7 alpha.1

shakkernerd committed:
Merge pull request #1361 from elizaOS/develop

chore: bump version to v0.1.7-alpha.1

actions-user committed:
chore: update changelog

shakkernerd committed:
Merge branch 'develop' into add-note-on-pnpm-node-version

shakkernerd committed:
Merge branch 'develop' into update-solana-sendtransaction-logic

shakkernerd committed:
Merge branch 'develop' into chore/twitter-cookies

proteanx committed:
add imageSettings types

proteanx committed:
pull imageSettings from character

proteanx committed:
use imageSettings from character file

while still retaining the ability to call via options with js

calintje commented:
Hi! I see that the integration tests are failing due to missing required API keys. Let me know if there's anything you need from my side!

shakkernerd commented:
Hello there, you don't have to do anything.

shakkernerd committed:
fix: explicitly set env in each step

shakkernerd committed:
Merge pull request #1374 from elizaOS/fix/integration_tests

fix: explicitly set env in each step

shakkernerd committed:
Merge branch 'develop' into update-solana-sendtransaction-logic

vkumar-dev committed:
Update README.md

adding documentation for running chat client

vkumar-dev committed:
Merge branch 'develop' into main

ryanleecode committed:
fix: postgres adapter settings not being applied

mcp97 committed:
Merge branch 'main' into realitySpiral/greetingworkflow

mcp97 committed:
Merge branch 'develop' into realitySpiral/closeStalPRsIssues

mcp97 committed:
Merge branch 'main' into realitySpiral/codescanning

mcp97 committed:
Merge pull request #1375 from onlyzerosonce/main

Update README.md to instructions to start client for chatting with Agent

mcp97 committed:
Merge branch 'develop' into add-note-on-pnpm-node-version

mcp97 committed:
Merge pull request #1350 from trbutler4/add-note-on-pnpm-node-version

docs: Add documentation on pnpm node version

ileana-pr committed:
sample plugin documentation

ileana-pr committed:
title update

shakkernerd committed:
chore: change CI trigger

shakkernerd committed:
Merge pull request #1387 from elizaOS/fix/integration_tests

chore: change CI trigger

shakkernerd committed:
Merge branch 'develop' into update-solana-sendtransaction-logic

shakkernerd committed:
chore: require approval for integration test step

shakkernerd committed:
Merge branch 'develop' into fix/integration_tests

shakkernerd committed:
Merge pull request #1388 from elizaOS/fix/integration_tests

chore: require approval for integration test step

shakkernerd committed:
Merge branch 'develop' into update-solana-sendtransaction-logic

shakkernerd committed:
chore: split tests

shakkernerd committed:
Merge branch 'develop' into fix/integration_tests

shakkernerd committed:
Merge pull request #1390 from elizaOS/fix/integration_tests

chore: split tests

shakkernerd committed:
Merge branch 'develop' into update-solana-sendtransaction-logic

shakkernerd committed:
Merge branch 'develop' into plugin-documentation

shakkernerd committed:
Merge pull request #1385 from ileana-pr/plugin-documentation

docs: sample plugin documentation

Commit:
Standardize boolean values and update .env file pattern

Replace YES/NO with true/false for boolean variables in .env file.
Ensure consistency across the codebase by removing the need for parseBooleanFromText.

proteanx committed:
Merge branch 'develop' into fixImageGen

jasonqindev committed:
Fix: Duplicate tweet scheduling loops in start method

- Resolved an issue where `generateNewTweetLoop` was called twice in the `start` method of /packages/client-twitter/src/post.ts.\n- The method was invoked at line 174 and line 190 without any conditional checks, leading to two independent tweet scheduling loops being created.\n- Removed one of the redundant `generateNewTweetLoop` calls to ensure only a single scheduling loop is initialized.\n\nFixes #1395

jasonqindev committed:
Merge branch 'main' into fix/duplicate-next-tweet-log

YoungPhlo committed:
docs: Add "What Did You Get Done This Week? 6" notes

jasonqindev committed:
Fix:  Move generateNewTweetLoop() into the if condition for enableActionProcessing

YoungPhlo committed:
docs: Standardize formatting for December 2024 streams

jasonqindev committed:
Merge branch 'develop' into fix/duplicate-next-tweet-log

mcp97 committed:
Merge branch 'develop' into realitySpiral/closeStalPRsIssues

mcp97 committed:
Merge branch 'develop' into realitySpiral/greetingworkflow

mcp97 committed:
Merge branch 'develop' into realitySpiral/codescanning

mcp97 committed:
Merge pull request #1399 from YoungPhlo/docs/community-streams

docs: Add "What Did You Get Done This Week? #6" notes

mcp97 committed:
Merge branch 'develop' into 1391-standardize-boolean-values-in-env-file

0xJord4n committed:
feat(github): add workflow to generate translations for README.md in multiple languages on push events

0xJord4n committed:
chore(workflow): update trigger for README translations to specific path to streamline workflow execution

0xJord4n committed:
chore(workflow): update branch name format in generate-readme-translations.yml to use quotes for consistency and prevent potential parsing issues

0xJord4n committed:
chore(workflow): update branch specification for push event in README translation workflow to follow proper YAML structure

mcp97 committed:
Merge pull request #1392 from hcaumo/1391-standardize-boolean-values-in-env-file

Standardize boolean values and update .env file pattern

mcp97 committed:
Merge branch 'develop' into fix/duplicate-next-tweet-log

mcp97 committed:
Merge pull request #1402 from jasonqindev/fix/duplicate-next-tweet-log

fix: duplicate tweet log

mcp97 committed:
Merge branch 'develop' into fix/postgres-adapter-settings

mcp97 committed:
Merge pull request #1379 from ryanleecode/fix/postgres-adapter-settings

fix: postgres adapter settings not being applied

mcp97 committed:
Merge branch 'develop' into fixImageGen

odilitime commented on packages/plugin-solana/src/actions/transfer.ts:
```diff
@@ -207,21 +207,43 @@ export default {
+                const transactionStartTime = Date.now();
+                const signature = await connection.sendTransaction(transaction, {
+                    maxRetries: 0,
+                    skipPreflight: true,
```
why are we skipping these?

calintje replied:
I optimized this implementation for transaction landing.

Many RPC-providers have their nodes behind loadbalancers. It might happen that when you fetch the latest blockhash (line 202), you get it from a node that is ahead of the one your transaction is being simulated on, resulting in a "blockhash not found" error. This is quite common.

https://www.helius.dev/blog/how-to-land-transactions-on-solana#how-do-i-land-transactions

calintje replied:
Let me know if you anything else from me

mcp97 committed:
Merge pull request #1371 from proteanx/fixImageGen

fix: image generation using imageSettings

0xJord4n committed:
chore(workflow): update aixion action version from v1.0.0 to v1.0.1 to incorporate latest improvements and fixes

ChristopherTrimboli committed:
Merge remote-tracking branch 'upstream/develop' into chore/twitter-cookies

mcp97 committed:
Merge branch 'develop' into realitySpiral/codescanning

proteanx committed:
venice image styles + watermark option

adds the ability to use venice's style presets as well as choose to remove the default watermark

mcp97 committed:
Merge pull request #1410 from proteanx/veniceImgFeats

feat: add venice style presets & option to remove watermark (image generation)

ryanleecode committed:
chore: format package.json files with prettier

ryanleecode committed:
chore: allow scoped pr titles

0xCardiE committed:
run multiple agents locally with auto setting of loclahost port

its-DeFine committed:
Merge branch 'develop' into bug-1270-incorrect-fallback

0xCardiE committed:
add message to exaplain how to use it

mcp97 committed:
Merge pull request #1414 from ryanleecode/chore/scoped-pr-titles

chore: allow scoped pr titles

twilwa committed:
Merge branch 'develop' into chore/format-package.json

mcp97 committed:
Merge branch 'develop' into realitySpiral/greetingworkflow

twilwa committed:
Merge pull request #1412 from ryanleecode/chore/format-package.json

chore: format package.json files with prettier

mcp97 committed:
Merge branch 'develop' into chore/twitter-cookies

mcp97 committed:
Merge pull request #1330 from ChristopherTrimboli/chore/twitter-cookies

fix: Twitter login notifications, incorrect cookie management.

mcp97 committed:
Merge branch 'develop' into multiple_localhosts

mcp97 committed:
Merge pull request #1415 from 0xCardinalError/multiple_localhosts

fix: Multiple Agents running at the same time on localhost

mcp97 committed:
Merge branch 'main' into tcm-compose-random-user

mcp97 committed:
Merge branch 'develop' into tcm-compose-random-user

mcp97 committed:
Merge branch 'develop' into bug-1270-incorrect-fallback

mcp97 committed:
Merge pull request #1305 from elizaOS/tcm-compose-random-user

fix: {{user}} tags in templates/examples empty when passed to LLM

mcp97 committed:
Merge branch 'develop' into update-env-plugin-goat

mcp97 committed:
Merge branch 'develop' into bug-1270-incorrect-fallback

mcp97 committed:
Merge pull request #1272 from UD1sto/bug-1270-incorrect-fallback

fix: fix imageModelProvider apiKey selection fallback

mcp97 committed:
Merge branch 'develop' into update-env-plugin-goat

mcp97 committed:
Merge pull request #1180 from aeither/update-env-plugin-goat

chore: update env for plugin-goat

harveyjavier committed:
Add Tagalog README markdown translation

harveyjavier committed:
Improve tagalog translation in

harveyjavier committed:
Add Tagalog README markdown translation

harveyjavier committed:
Improve tagalog translation in

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1420 from harveyjavier/main

docs: Add Tagalog README Translation

shakkernerd committed:
chore: Update cron schedule to run every Saturday at 8:29 AM

shakkernerd committed:
Merge branch 'develop' into realitySpiral/codescanning

shakkernerd committed:
Merge pull request #1314 from elizaOS/realitySpiral/codescanning

feat: [Code Scanning] Security Improvements - create codeql.yml

shakkernerd committed:
Merge branch 'develop' into realitySpiral/greetingworkflow

shakkernerd committed:
Merge pull request #1316 from elizaOS/realitySpiral/greetingworkflow

feat: greet first time contributors

shakkernerd committed:
Merge branch 'develop' into realitySpiral/closeStalPRsIssues

shakkernerd committed:
Merge pull request #1317 from elizaOS/realitySpiral/closeStalPRsIssues

feat: add auto PR / issue close after being stale for a certain amount of time

omahs committed:
fix typos

omahs committed:
fix typo

RiceChuan committed:
docs: 1.Quotation marks are used incorrectly.2.Delete duplicate words

Signed-off-by: RiceChuan <lc582041246@gmail.com>

tomguluson92 committed:
Update index.ts

yerinle committed:
Update eliza-in-tee.md (fixing typo)

0xJord4n committed:
chore(workflow): update aixion action version from v1.0.1 to v1.0.2 for improved functionality and bug fixes

alessandromazza98 committed:
improve logging in plugin-coinbase

0xJord4n committed:
chore(generate-readme-translations): escape quotes and special characters in README content to ensure proper formatting during translation process

alessandromazza98 committed:
Merge branch 'develop' into 1194-improve-logging-in-plugins-coinbase

0xJord4n committed:
fix(workflow): improve translation saving process by escaping special characters to ensure correct file creation

0xJord4n committed:
chore(generate-readme-translations): refactor workflow to simplify README translation process and improve artifact handling

- Remove unnecessary steps for reading and escaping README content.
- Update the action version for translation to v1.2.0 for better performance.
- Use content_path to directly reference README.md for translation.
- Add step to upload each translated file as an artifact.
- Introduce a new commit step to handle all translated README files at once.

0xJord4n committed:
chore(workflow): update aixion action version from v1.2.0 to v1.2.1 for improved functionality and bug fixes

0xJord4n committed:
Merge branch 'elizaOS:main' into 1222--README-ci-auto-translation

nulLeeKH committed:
fix: make twitter search feature works

nulLeeKH committed:
doc: add .idea in .gitignore

nulLeeKH committed:
Merge branch 'main' of github.com:nulLeeKH/eliza

nulLeeKH committed:
doc: modify some comments

nulLeeKH committed:
Merge branch 'develop' of github.com:elizaOS/eliza

fyInALT committed:
add only to booleanFooter

xwxtwd committed:
feat: plugin-evm added AlienXChain.

fyInALT committed:
Merge branch 'develop' into fix/add-a-limit-to-generate-loop

mcp97 committed:
Merge pull request #1437 from fyInALT/fix/add-a-limit-to-generate-loop

feat: add `only` to booleanFooter

mcp97 committed:
Merge branch 'develop' into 1194-improve-logging-in-plugins-coinbase

mcp97 committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1429 from alessandromazza98/1194-improve-logging-in-plugins-coinbase

improve logging in plugin-coinbase

mcp97 committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1428 from yerinle/patch-1

Update eliza-in-tee.md (fixing typo)

mcp97 committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1423 from omahs/patch-1

fix: typos

mcp97 committed:
Merge branch 'develop' into main

0xJord4n committed:
Merge branch 'develop' into 1222--README-ci-auto-translation

mcp97 committed:
Merge pull request #1424 from RiceChuan/main

docs: 1.Quotation marks are used incorrectly.2.Delete duplicate words

Commit:
fix: use elizaos package

nulLeeKH committed:
mod: replace mistypped log in twitter search log

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge branch 'develop' into patch-3

mcp97 committed:
Merge pull request #1425 from tomguluson92/patch-3

feat: client-github retry

nulLeeKH committed:
rfc: accept pr feedback

nulLeeKH committed:
Merge branch 'develop' of github.com:elizaOS/eliza

nulLeeKH committed:
Merge branch 'main' of github.com:nulLeeKH/eliza

mcp97 committed:
Merge branch 'develop' into feat/add_alienx_chain

mcp97 committed:
Merge pull request #1438 from xwxtwd/feat/add_alienx_chain

feat: (plugin-evm) add alienx chain

cygaar committed:
Rebase

cygaar committed:
Get builds working

cygaar committed:
Get eth transfers working

0xJord4n committed:
Merge branch 'develop' into 1222--README-ci-auto-translation

mcp97 committed:
Merge pull request #1432 from 0xJord4n/1222--README-ci-auto-translation

chore: Keeps README translations synchronized

mcp97 committed:
Merge branch 'develop' into main

cygaar committed:
Add erc20 support

cygaar committed:
Add default abstract rpc

shakkernerd committed:
Merge pull request #1225 from cygaar/abstract_plugin

feat: add abstract plugin

twilwa committed:
Merge branch 'develop' into main

twilwa committed:
Merge pull request #1433 from nulLeeKH/main

fix: Make search feature in twitter client works

mhxw committed:
fix: fix incorrect link redirection issue

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1443 from mhxw/main

fix: fix incorrect link redirection issue

odilitime committed:
Merge branch 'develop' into update-solana-sendtransaction-logic

calintje commented:
Hi @odilitime,

I'm working at Orca and am currently working on a new plugin for Eliza that should automate LP repositioning. I've already implemented this as a regular bot using our SDKs, but for Eliza I want to leverage the provider-evaluator loop. I got the provider and actions working, but would love a little nudge into the right direction to leverage the Evaluator the right way.

I think a 10-15 minute screenshare session would be sufficient.

With Orca, we're also in a TG chat with ai16z folks, but I find it really hard to get in touch with devs. Would appreciate you helping out a bit 🙏🏼

Freytes commented:
> Hi @odilitime,
> 
> I'm working at Orca and am currently working on a new plugin for Eliza that should automate LP repositioning. I've already implemented this as a regular bot using our SDKs, but for Eliza I want to leverage the provider-evaluator loop. I got the provider and actions working, but would love a little nudge into the right direction to leverage the Evaluator the right way.
> 
> I think a 10-15 minute screenshare session would be sufficient.
> 
> With Orca, we're also in a TG chat with ai16z folks, but I find it really hard to get in touch with devs. Would appreciate you helping out a bit 🙏🏼

I can help with this as well.

odilitime commented:
Hello,

We are changing our plugin development strategy to be more scalable. We have moved the plugins out into their own repos and we're looking for people to either maintain those or own them on their own Github.

Please make your code changes to this repo: https://github.com/elizaos-plugins/plugin-solana

If you'd like to be a maintainer,  file an issue in the plugin repo and join our Discord https://discord.gg/elizaos to coordinate.

If you'd like to control the plugin on your own Github, please add an issue to the plugin repo pointing to your repo, and add a modification to the registry. Submit a PR to edit the registry here: https://github.com/elizaos-plugins/registry

Closing this PR for now. Let us know if you have any questions
